CPDShell/Core/algorithms/ClassificationBasedCPD/abstracts/iclassifier.py

```diff
@@ -0,0 +1,33 @@
+"""
+Module for Classification CPD algorithm's classifier abstract base class.
+"""
+
+__author__ = "Artemii Patov"
+__copyright__ = "Copyright (c) 2024 Artemii Patov"
+__license__ = "SPDX-License-Identifier: MIT"
+
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+
+import numpy as np
+
+
+class Classifier(ABC):
+    """Classifier's abstract base class."""
+
+    @abstractmethod
+    def classify(self, window: Iterable[float | np.float64]) -> None:
+        """Applies classificator to the given sample.
+
+        :param window: part of global data for finding change points.
+        """
+        raise NotImplementedError
```

Comment:
Why throw such an error in an abstract method? 🦣

Reply:
It tells users that they definetely should override it.